docs/docs/tutorial/creating-a-source-plugin/part-2/index.mdx

```diff
@@ -0,0 +1,682 @@
+- The strict 1:1 mapping of API endpoints to node types would mean you'd have `Author` and `AuthorDetail`
+- However, you shouldn't create `Author` and `AuthorDetail` node types but instead only a `Author` one. While creating the `Author` one, access the information from the author detail endpoint and enrich the `Author` information.
+
+<Announcement>
+
+**Need a refresher on what "nodes" are?** Don't worry!
+
+Inside the Gatsby's data layer, information is stored in objects called **nodes**. A node is the smallest form unit of data in the data layer. Different source plugins create different types of nodes, each of which have their own properties. For example, `gatsby-source-filesystem` creates `File` nodes.
+
+Also be sure to checkout the [GraphQL concepts](/docs/conceptual/graphql-concepts/) page if you need more information.
+
+</Announcement>
+
+### `createNode`
+
+Besides fetching of the actual data, the creation of nodes is another important piece to a source plugin.
+
+<Announcement>
+
+**Key Gatsby Concept** 💡
+
+Be sure to read this section carefully as understanding how [`createNode`](/docs/reference/config-files/actions/#createNode) works is really important for the rest of this tutorial and for source plugins in general.
+
+</Announcement>
+
+A basic usage of `createNode` looks like this:
+
+```ts:title=gatsby-node.ts
+import type { GatsbyNode, NodeInput } from "gatsby"
+
+export const sourceNodes: GatsbyNode[`sourceNodes`] = (gatsbyApi) => {
+  // Your arbitrary, serializable data
+  const yourData = {
+    hello: `world`,
+  }
+
+  const node = {
+    ...yourData,
+    // Required fields
+    id: gatsbyApi.createNodeId(`some-string`),
+    internal: {
+      type: `TypeName`,
+      contentDigest: gatsbyApi.createContentDigest(yourData)
+    }
+  } satisfies NodeInput
+
+  gatsbyApi.actions.createNode(node)
+}
+```
+
+<Collapsible
+  summary={<em>Explanation on each required node field</em>}
+>
+
+Here's a short explanation on each required field for your convenience:
+
+- `id`: The node's ID that Gatsby uses to track it. It must be a globally unique ID. You should use the [`createNodeId`](/docs/reference/config-files/node-api-helpers/#createNodeId) helper to create it.
+- `internal`: An object that only allows a defined set of keys (see API reference)
+  - `type`: The name of the node type
+  - `contentDigest`: A hash (or short digital summary) of the contents of a node. The digest should be unique to the node since it's used for caching. If the node changes, this digest should also change. You can use the [`createContentDigest`](/docs/reference/config-files/node-api-helpers/#createContentDigest) helper to create that hash.
+  
+    If your node contains information like a revision ID or an `lastUpdated` timestamp, you should use this instead of creating your own content digest. This will lead to lower CPU utilization and still achieve the goal of invalidating the cache for the node when said node changes.
```

Comment:
Small typo:
```suggestion
    If your node contains information like a revision ID or a `lastUpdated` timestamp, you should use this instead of creating your own content digest. This will lead to lower CPU utilization and still achieve the goal of invalidating the cache for the node when said node changes.
```